lib/reek/context_builder.rb

```diff
@@ -465,8 +473,26 @@ def inside_new_context(klass, exp)
     # @return [Context::*Context] - the context that was appended
     #
     def append_new_context(klass, *args)
-      klass.new(element, *args).tap do |new_context|
-        element.append_child_context new_context
+      klass.new(current_context, *args)
+    end
+
+    def handle_send_for_modules(exp)
+      method_name = exp.method_name
+      arg_names = exp.arg_names
+      current_context.track_visibility(method_name, arg_names)
+      current_context.track_singleton_visibility(method_name, arg_names)
+      register_attributes(exp) if exp.attribute_writer?
```

Comment:
How about moving the check for `exp.attribute_writer?` into the method `register_attributes` ?